### PR TITLE
add tests for `util/Travis`

### PR DESCRIPTION
**Pull request by jacobroufa**

### CODE REVIEW

File: tests/unit/src/util/Travis.ts [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+					repo.findAuthorization.reset();
+				},
+
+				async 'existing authorization; eventually throws'() {
+					repo.findAuthorization.returns(Promise.resolve({ id: 1 }));
+
+					try {
+						await travis.createAuthorization(repo);
+
+						assert.fail();
```
Comment (devpaul) [inline tests/unit/src/util/Travis.ts]: Small nit-pick. The catch message will catch this and only fail on the error message assertion. We should explicitly fail it before it reaches the strict equal to make this test slightly better.

File: tests/unit/src/util/Travis.ts [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+					auth.restore();
+					deleteAuth.restore();
+				},
+
+				async 'authentication fails; eventually throws'() {
+					const auth = stub(travis, 'authenticate').returns(Promise.reject());
+					const deleteAuth = stub(travis, 'deleteAuthorization').returns(Promise.resolve());
+
+					try {
+						await travis.createAuthorization(repo);
```
Comment (devpaul) [inline tests/unit/src/util/Travis.ts]: This test passes if `#createAuthorization()` doesn't throw. We should test for that case as well.

File: tests/unit/src/util/Travis.ts (hunk 1 above)
Comment (jacobroufa) [inline tests/unit/src/util/Travis.ts]: Addressed in 53ff3f4. Thanks for clarifying this one!

File: tests/unit/src/util/Travis.ts (hunk 2 above)
Comment (jacobroufa) [inline tests/unit/src/util/Travis.ts]: addressed in 53ff3f4

**Pull request merged**